File: 2840-check-if-strings-can-be-made-equal-with-operations-ii/2840-check-if-strings-can-be-made-equal-with-operations-ii.cpp
class Solution {
public:
    bool checkStrings(string s1, string s2) {
        int n = s1.length();

        if (s1 == s2) return true;

        string evenStr = "";
        string oddStr = "";

        string evenStr2 = "";
        string oddStr2 = "";

        for (int i = 0; i < n; i++) {
            if (i % 2 == 0) {
                evenStr  += s1[i];
                evenStr2 += s2[i];
            }
            else {
                oddStr  += s1[i];
                oddStr2 += s2[i];
            }
        }

        sort(evenStr.begin(), evenStr.end());
        sort(evenStr2.begin(), evenStr2.end());
        sort(oddStr.begin(), oddStr.end());
        sort(oddStr2.begin(), oddStr2.end());

        return (evenStr == evenStr2) && (oddStr == oddStr2);

    }
};

/*
which numbers difference will be even -> (odd - odd), (even - even)

abcdba
even - a, c, b
odd  - b, d, a

cabdab
even - c, b, a
odd  - a, d, b 
*/

File: 2840-check-if-strings-can-be-made-equal-with-operations-ii/2840-check-if-strings-can-be-made-equal-with-operations-ii.cs
public class Solution {
    public bool CheckStrings(string s1, string s2) {
        int n = s1.Length;

        int[] even = new int[26];
        int[] odd  = new int[26];

        for (int i = 0; i < n; i++) {
            if (i%2 == 0) {
                even[s1[i] - 'a']++;
                even[s2[i] - 'a']--;
            }
            else {
                odd[s1[i] - 'a']++;
                odd[s2[i] - 'a']--;
            }
        }

        for (int i = 0; i < 26; i++) {
            if (even[i] != 0 || odd[i] != 0)
                return false;
        }
        return true;
    }
}